windtools/vtk.py

```diff
@@ -0,0 +1,339 @@
+        # scalar
+        ds = xr.Dataset({'var':(('x','y'), out[0,:])})
+    elif len(out) == 6:
+        # tensor
+        ds = xr.Dataset({'var_xx':(('x','y'), out[0,:]), 'var_xy':(('x','y'), out[1,:]), 'var_xz':(('x','y'), out[2,:]), \
+                         'var_yy':(('x','y'), out[3,:]), 'var_yz':(('x','y'), out[4,:]), 'var_zz':(('x','y'), out[5,:])})
+    else:
+        raise ValueError(f'Read variable with {len(out)} components. Not sure how to follow. Stopping.')
+        
+    ds = ds.assign_coords({'x':x, 'y':y})
+    
+    if isinstance(t, (str, int, float)):
+        if type(dateref) in (datetime, datetime.datetime, pd.Timestamp):
+            t =  pd.to_datetime(float(t), unit='s', origin=dateref).round('0.1S')
+            timename = 'datetime'
+        else:
+            t = float(t)
+            timename = 'time'
+        ds[timename] = t
+        ds = ds.expand_dims(timename).assign_coords({timename:[t]})
+    
+    return ds
+    
+    
+    
+def readSingleVTK(vtkfullpath, res=10, squash=None):
+    """
+    Read a single VTK file and interpolates into a uniform grid based on the limits read from the VTK
+    
+    Parameters:
+    -----------
+    vtkfullpath: str
+        full path and filename of the desired VTK file
+    res: int, float
+        resolution of interpolated grid
+    squash: str, 'x', 'y', or 'z' only
+        Squash the VTK into a plane (useful for terrain slices)
+    
+    Returns:
+    --------
+    x, y:
+        meshgrid of x and y positions.
+    out:
+        array of shape (<numComponents>, <nx>, <ny>), where x and y are the long and short 
+        direction, respectively. E.g. for Uz, out[2]
+    
+    """
+    
+    # Check if the slice actually exists
+    if not os.path.isfile(vtkfullpath):
+        raise FileNotFoundError(f'Slice {os.path.basename(os.path.dirname(vtkfullpath))}/' \
+                                f'{os.path.basename(vtkfullpath)} does not exist.')
+    
+    # Open VTK
+    reader = vtk.vtkPolyDataReader()
+    reader.SetFileName(vtkfullpath)
+    reader.ReadAllVectorsOn()
+    reader.Update()
+    
+    # Load data
+    polydata = dsa.WrapDataObject(reader.GetOutput())
+    ptdata   = polydata.GetPointData()
+    coords   = np.round(polydata.Points, 4)
+    data     = ptdata.GetArray(0)
+
+    # Determine what plane the VTK is in
+    if squash == None:
+        # There is a bug with OpenFOAM where a slice in a certain plane gets lots of points also on planes
+        # x=0 or y=0. Here we check the variability of points in all 3 directions and get the direction w/
+        # with less. Example: a slice that is supposed to be at x=1000, will also have of points on x=0
+        nPointsPerDir = [len(np.unique(coords[:,i])) for i in [0,1,2]]
+        dir1 = np.argsort(nPointsPerDir)[2] # the long direction
+        dir2 = np.argsort(nPointsPerDir)[1] # the short direction
+        dirconst = np.argsort(nPointsPerDir)[0]  # 0:xNormal, 1:yNormal, 2:zNormal
```

Comment:
Default behavior should be to return 3D box instead of guessing a squash direction

Reply:
Resolved by commit efddd1e305291d79e4226766d4aa96e2cd44d076